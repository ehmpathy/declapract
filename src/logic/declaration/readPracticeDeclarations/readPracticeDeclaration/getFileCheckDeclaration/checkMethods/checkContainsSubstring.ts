import { diff } from 'jest-diff';

import { UnexpectedCodePathError } from '../../../../../UnexpectedCodePathError';

export const checkContainsSubstring = ({
  declaredContents,
  foundContents,
  bAnnotation = 'Received',
}: {
  declaredContents: string;
  foundContents: string;
  bAnnotation?: string;
}) => {
  try {
    expect(foundContents).toContain(declaredContents);
  } catch (error) {
    // if the above check failed, run diff on the string directly to show a better string diff message
    const difference = diff(declaredContents, foundContents, { aAnnotation: 'Expected toContain', bAnnotation });
    if (!difference)
      throw new UnexpectedCodePathError(
        'expect().toContain() threw an error, but no difference was detected in the strings',
      );
    throw new Error(difference);
  }
};
