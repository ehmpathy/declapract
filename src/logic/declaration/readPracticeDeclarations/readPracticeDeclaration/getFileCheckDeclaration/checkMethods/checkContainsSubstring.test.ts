import { checkContainsSubstring } from './checkContainsSubstring';

describe('checkContainsSubstring', () => {
  it('should return nothing when the string is a substring', () => {
    const result = checkContainsSubstring({
      declaredContents: 'a-substring',
      foundContents: 'a-big-string with a-substring inside',
    });
    expect(result).not.toBeDefined();
  });
  it('should throw an error with a good looking diff when not equal', () => {
    try {
      checkContainsSubstring({
        declaredContents: 'a-substring',
        foundContents: 'a-big-string with a-different-substring inside',
      });
      fail('should not reach here');
    } catch (error) {
      expect(error.message).toContain('toContain');
      expect(error.message).toMatchSnapshot(); // log example of it
    }
  });
});
