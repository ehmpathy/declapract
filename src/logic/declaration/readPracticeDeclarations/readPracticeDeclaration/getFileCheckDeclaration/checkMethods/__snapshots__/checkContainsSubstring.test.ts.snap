// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkContainsSubstring should throw an error with a good looking diff when not equal 1`] = `
"[32m- Expected toContain[39m
[31m+ Received[39m

[32m- a-substring[39m
[31m+ a-big-string with a-different-substring inside[39m"
`;
